SFUsat/sfu_err_tab.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * sfu_err_tab.h
+ *
+ * Error table structure to be stored in Error Log Queue
+ *
+ * Created on:
+ * 	   Author: Seleena
+ * */
+
+#ifndef SFUSAT_ERR_TAB_H
+#define SFUSAT_ERR_TAB_H
+
+
+#include <sys/time.h> //memory concerns for the embedded system?
+#include <string.h>
+
+#define desc_limit 20 //size limit for description in ErrTab_t
+
+typedef enum { //change error types later
+	ER1,
+	ER2,
+	ER3
+
+} ErrType_t;
+
+typedef struct {
+
+	struct timeval timestamp;
+	ErrType_t error_type; //Change later
+	char* description;
```

Comment:
It will be useful indeed to be able to attach descriptions to errors. However, this would mean the use of dynamic memory in the case of arbitrary strings, or a big waste of memory in the case of static allocations. 

Perhaps it'd be better to have enough unique and descriptive `ErrType_t`s so that for the majority of entries, they won't need to set a unique description. 

For the ones that do, we could create a separate array of character arrays and then have each `ErrTab_t` keep track of this index in a small variable (eg, a `char` or `uint8_t`). 

This way, we could have more `ErrTab_t`s statically declared and have less memory wasted, while also allowing for a certain amount of `ErrTab_t`s to attach a description.